packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -80,11 +101,18 @@ class FlutterPlugin implements Plugin<Project> {
         return result
     }
 
-    private String getTargetPlatform(Project project) {
+    private Set<String> getTargetPlatforms(Project project) {
         if (project.hasProperty('target-platform')) {
-            return project.property('target-platform')
+            Set<String> platforms =  []
```

Comment:
This is another very subjective nit, feel free to ignore.

If you flip the structure of this into an early return you end up with most of the logic being outside of a block instead of inside of one, so you don't have to keep as much nesting in mind while you're reading the function and trying to figure out the control flow. Subjective but it helps readability a bit in my opinion. This applies in a couple places.

```groovy
if (!project.hasProperty('target-platform')) {
  return DEFAULT_PLATFORMS
}
Set<String> platforms = []; 
project.property('target-platform').split(',').each { // etc
```

https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Also a really minor optional nit, but I think mapping here would be a little cleaner than doing a for-each loop and then appending to a list. The groovy version of map is [`collect()`](http://docs.groovy-lang.org/2.4.3/html/groovy-jdk/java/util/Collection.html#collect(groovy.lang.Closure)).

Reply:
+1